java/core/src/main/java/sleeper/core/util/ByteArray.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * ------------------
+ * Copyright (C) 2012 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.util;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * Code utilised below is a subset of the ByteArray class from facebook.collections.
+ *
+ * A wrapper object for primitive byte array type.
+ * Required as hashCode and equals methods required for usage of type within a HashSet.
+ */
+public class ByteArray implements Comparable<ByteArray> {
+
+    final ByteArrayComparator byteArrayComparator = new ByteArrayComparator();
+
+    private final byte[] array;
+
+    ByteArray(byte[] array) {
+        this.array = array;
+    }
+
+    /**
+     * Returns the contents of this object as a primitive byte array.
+     *
+     * @return the byte array
+     */
+    public byte[] getArray() {
```

Comment:
Please apply our conventions for method ordering:

https://github.com/gchq/sleeper/blob/develop/docs/development/conventions.md#ordering-within-a-java-class